dashboard/app/models/levels/blockly.rb

```diff
@@ -379,6 +379,35 @@ def localized_blockly_level_options(script)
     options.freeze
   end
 
+  def localize_blockly_level_options_for_lab2(script)
+    options = Rails.cache.fetch("#{cache_key}/#{script.try(:cache_key)}/#{I18n.locale}/localized_blockly_level_options", force: !Unit.should_cache?) do
+      level_options = blockly_level_options.dup
+      set_unless_nil(level_options, 'sharedBlocks', localized_shared_blocks(level_options['sharedBlocks']))
+
+      functions = level_options.
+        try(:[], "levelData").
+        try(:[], "startSources").
+        try(:[], "blocks").try(:[], "blocks")&.
+        filter {|block| block.try(:[], "type") == "procedures_defnoreturn"}
+
+      functions.each do |function|
+        function_name = function.try(:[], "fields").try(:[], "NAME")
+        next unless function_name
+
+        localized_name = I18n.t(
+          "name",
+          scope: [:data, :function_definitions, name, function_name],
+          default: nil,
+          smart: true
+        )
+
+        function["fields"]["NAME"] = localized_name
+      end
+    end
+
+    options.freeze
+  end
```

Comment:
This function is based on`localized_blockly_level_options`, immediately above.  However, it doesn't seem necessary to actually return anything, because the modifications appear to occur in-place.